src/test/java/programminglife/gui/GuiControllerTest.java

```diff
@@ -1,67 +1,234 @@
 package programminglife.gui;
 
-import javafx.application.Platform;
-import javafx.fxml.FXMLLoader;
-import javafx.scene.Node;
-import javafx.scene.Parent;
-import javafx.scene.Scene;
-import javafx.scene.control.Button;
+import javafx.scene.input.KeyCode;
 import javafx.stage.Stage;
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Test;
-import org.testfx.framework.junit.ApplicationTest;
+import org.junit.*;
+import org.testfx.api.FxRobot;
+import org.testfx.api.FxToolkit;
 import org.testfx.util.WaitForAsyncUtils;
-import programminglife.model.exception.UnknownTypeException;
+import programminglife.ProgrammingLife;
 
 import java.io.File;
-import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
+import static org.junit.Assert.assertTrue;
+
 /**
- * Created by Yannick on 12/05/2017.
+ * This test class is there to interactively test the GUI. It is capable of clicking on certain buttons and items
+ * that are present within the GUI. It is important however not to move the mouse during this process!!!
+ * This is only usable if you have a USA QWERTY layout on your keyboard!!!
  */
-public class GuiControllerTest extends ApplicationTest {
-    private Button btnZoomIn;
-    private Button btnZoomOut;
-    private Button btnZoomReset;
-    private Button btnDrawRandom;
-    private FXMLLoader loader = new FXMLLoader();
+public class GuiControllerTest extends FxRobot {
+    private static Stage primaryStage;
+    private ProgrammingLife pl;
+    private String f;
+    private static String OS = System.getProperty("os.name").toLowerCase();
 
-    @Override public void start(Stage stage) throws IOException, UnknownTypeException {
-        Parent p = FXMLLoader.load(getClass().getResource("/Basic_Gui.fxml"));
-        stage.setScene(new Scene(p));
-        stage.show();
-        stage.toFront();
-        File f = new File(getClass().getResource("/test.gfa").getPath());
+    public GuiControllerTest() {
+        try {
+            f = new File(getClass().getResource("/test.gfa").toURI()).getAbsolutePath();
+        } catch (URISyntaxException e) {
+            f = null;
+        }
     }
 
-    private <T extends Node> T find(final String query) {
-        return lookup(query).queryFirst();
+    @BeforeClass
+    public static void setUpClass() {
+        try {
+            primaryStage = FxToolkit.registerPrimaryStage();
+        } catch (TimeoutException e) {
+            e.printStackTrace();
+        }
     }
 
     @Before
     public void setUp() {
-        btnZoomIn = find("#btnZoomIn");
-        btnZoomOut = find("#btnZoomOut");
-        btnZoomReset = find("#btnZoomReset");
-        btnDrawRandom = find("#btnDrawRandom");
+        try {
+            this.pl = (ProgrammingLife) FxToolkit.setupApplication(ProgrammingLife.class);
+            WaitForAsyncUtils.waitFor(10, TimeUnit.SECONDS, primaryStage.showingProperty());
```

Comment:
There should be a better method to assure the window is open. Also, 10 seconds is quite a long wait when doing a lot of testing.